app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java

```diff
@@ -961,6 +981,10 @@ public void handleResult(@NonNull final SearchInfo result) {
                     NewPipe.getNameOfService(serviceId), searchString, 0);
         }
 
+        handleSearchSuggestion(result.getSearchSuggestion(), result.isCorrectedSearch());
+
+        searchSuggestionString = result.getSearchSuggestion();
+        isCorrectedSearch = result.isCorrectedSearch();
```

Comment:
why do you save it *after* calling handleSearchSuggestion()? here you call result.getSearchSuggestion() and result.isCorrectedSearch() twice.